src/views/highchart.js

```diff
@@ -0,0 +1,169 @@
+
+        this._series = {};
+
+        this._seriesDetector = new SeriesDetector({
+            fieldsToIgnore: [ this._attributes.xField ],
+            keyField: this._attributes._keyField
+        });
+
+        // Inject this view's el into the user-provided
+        // highchart constructor options.
+        let highchartConfig = _.extend(options.highchartOptions, {
+            chart: _.extend({ renderTo: this.sinkBodyEl }, options.highchartOptions.chart )
+        });
+
+        this.chart = Highcharts.chart(highchartConfig);
+    }
+
+    _applyOptionDefaults(options) {
+        options = options || {};
+
+        return _.extend({
+            xField: 'time',
+            highchartOptions: this._applyHighchartOptionsDefaults(options.highchartOptions)
+        }, options);
+    }
+
+    _applyHighchartOptionsDefaults(highchartOptions) {
+        highchartOptions = highchartOptions || {};
+
+        return _.extend({
+            xAxis: _.extend({
+                type: 'datetime'
+            }, highchartOptions.xAxis),
+            chart: _.extend({}, highchartOptions.chart)
+        }, highchartOptions);
+    }
+
+    _addSeries(series) {
+        let seriesLabel = this._seriesDetector.getSeriesLabel(series.id);
+        let config = this._findMatchingSeriesConfig(series.keys);
+        seriesLabel = seriesLabel !== '' ? seriesLabel : this._attributes.valueField;
+
+        this.chart.addSeries(_.extend({
+            id: 'series-' + series.id,
+            name: seriesLabel
+        }, config));
+
+        this._series[series.id] = series;
+    }
+
+    _convertUserOptionsToAttributes(options) {
+        return _.pick(options, 'valueField', 'keyField', 'xField', 'series');
+    }
+
+    _consume(points) {
+        if (points.length === 0) {
+            return;
+        }
+
+        if (!this._attributes.valueField) {
+            this._attributes.valueField = this._findValueField(points[0]);
+        }
+
+        points.forEach((point) => {
+            let series = this._seriesDetector.getSeriesForPoint(point);
+
+            if (!this._series[series.id]) {
+                this._addSeries(series);
+            }
+
+            let xFieldValue = point[this._attributes.xField];
+
+            this.chart.get('series-' + series.id).addPoint([
+                _.isDate(xFieldValue) ? xFieldValue.getTime() : xFieldValue,
+                point[this._attributes.valueField] ]);
+        });
+    }
+
+    _findMatchingSeriesConfig(keys) {
+        let keyValues = _.values(keys);
+        let seriesConfig = {};
+        if (this._attributes.series) {
+            let seriesConfigKeys = Object.keys(this._attributes.series);
+            for(let i = 0; i < seriesConfigKeys.length; i++) {
+                let configKey = seriesConfigKeys[i];
+                let config = this._attributes.series[configKey];
+
+                if (_.contains(keyValues, configKey)) {
+                    seriesConfig = config;
```

Comment:
They don't go into the same series, the same user-provided configuration gets applied to the series. The situation you described isn't ideal but this is the way it works in `timechart` and this hasn't been an issue. Its most likely because different fields having the same values is a rare occurrence.

To get around the issue you present, we could change the series config to match against a series via a _key_ and _value_:

```
[
  {
    key: {
      host: 'A'
    },
    config: {
      color: ...
    }
  }
]
```

but that is too verbose for the common case.

At some point in `timechart` we only allowed series configs when a `keyField` was specified meaning that there was only one possible field to match the value against. However, this didn't work for situations where you had two keyFields, say `host` and `metricName` and you wanted a series config to apply to all series that had a particular `metricName`.